cmake/test/gtest.cmake

```diff
@@ -85,97 +85,261 @@ function(catkin_add_executable_with_gtest target)
+        set(GMOCK_FROM_SOURCE_INCLUDE_DIRS ${gmock_include_dir} CACHE INTERNAL "")
+        set(GMOCK_FROM_SOURCE_LIBRARY_DIRS ${gmock_lib_dir} CACHE INTERNAL "")
+        set(GMOCK_FROM_SOURCE_LIBRARIES ${gmock_libs} CACHE INTERNAL "")
+        set(GMOCK_FROM_SOURCE_MAIN_LIBRARIES ${gmock_main_libs} CACHE INTERNAL "")
+
+        set(_CATKIN_SKIP_INSTALL_RULES TRUE)
+        function(install)
+          if(_CATKIN_SKIP_INSTALL_RULES)
+            return()
+          endif()
+          _install(${ARGN})
+        endfunction()
+        add_subdirectory(${gmock_base_dir} ${gmock_lib_dir})
+        set(_CATKIN_SKIP_INSTALL_RULES FALSE)
+
+        set_target_properties(${gmock_libs} ${gmock_main_libs}
+                              PROPERTIES EXCLUDE_FROM_ALL 1)
+
+        message(STATUS "Found gmock sources under '${gmock_base_dir}': gtests will be built")
+      else() # gmock not found-- look for system-installed gtest by itself
+        set(_include_paths "/usr/include/gtest")
+        if(CATKIN_TOPLEVEL)
+          # search in the current workspace before
+          list(INSERT _include_paths 0 "${CMAKE_SOURCE_DIR}/gtest/include/gtest")
+        endif()
+
+        set(_source_paths "/usr/src/gtest/src")
+        if(CATKIN_TOPLEVEL)
+          # search in the current workspace before
+          list(INSERT _source_paths 0 "${CMAKE_SOURCE_DIR}/gtest/src")
+        endif()
+
+        catkin_find_gtest_source("${_include_paths}" "${_source_paths}" gtest_found
+                               gtest_base_dir gtest_include_dir gtest_lib_dir
+                               gtest_libs gtest_main_libs)
+
+        if(gtest_found)
+          set(GTEST_FROM_SOURCE_FOUND ${gtest_found} CACHE INTERNAL "")
+          set(GTEST_FROM_SOURCE_INCLUDE_DIRS ${gtest_include_dir} CACHE INTERNAL "")
+          set(GTEST_FROM_SOURCE_LIBRARY_DIRS ${gtest_lib_dir} CACHE INTERNAL "")
+          set(GTEST_FROM_SOURCE_LIBRARIES ${gtest_libs} CACHE INTERNAL "")
+          set(GTEST_FROM_SOURCE_MAIN_LIBRARIES ${gtest_main_libs} CACHE INTERNAL "")
+
+          set(_CATKIN_SKIP_INSTALL_RULES TRUE)
+          function(install)
+            if(_CATKIN_SKIP_INSTALL_RULES)
+              return()
+            endif()
+            _install(${ARGN})
+          endfunction()
+          add_subdirectory(${gtest_base_dir} ${gtest_lib_dir})
+          set(_CATKIN_SKIP_INSTALL_RULES FALSE)
+          set_target_properties(${gtest_libs} ${gtest_main_libs}
+                                PROPERTIES EXCLUDE_FROM_ALL 1)
+
+          message(STATUS "Found gtest sources under '${gtest_base_dir}': gtests will be built")
+        else()
+          if(CATKIN_TOPLEVEL)
+            message(STATUS "gtest not found, C++ tests can not be built. Please install the gtest headers globally in your system or checkout gtest (by running 'svn checkout http://googletest.googlecode.com/svn/tags/release-1.6.0 gtest' in the source space '${CMAKE_SOURCE_DIR}' of your workspace) to enable gtests")
```

Comment:
Should this reference at least version 1.7 or even newer and the GitHub repo instead?

Reply:
Fixed, it references 1.8 in the github repo.